File: lib/kotlin-mpp/src/commonMain/kotlin/com/sdercolin/utaformatix/data/TimeSignature.kt
package com.sdercolin.utaformatix.data

import kotlinx.serialization.Serializable

/**
 * Time signature model
 * @param measurePosition Measure (bar) position of the time signature
 * @param numerator Beats per measure
 * @param denominator Note value per beat
 */
@Serializable
data class TimeSignature(
    val measurePosition: Int,
    val numerator: Int,
    val denominator: Int
)

File: lib/csharp/UtaFormatix.Data/TimeSignature.cs
namespace UtaFormatix.Data;

/// <summary>Time signature model.</summary>
///
/// <param name="MeasurePosition">Measure (bar) position of the time signature.</param>
/// <param name="Numerator">Beats per measure.</param>
/// <param name="Denominator">Note value per beat.</param>
public record TimeSignature(int MeasurePosition, int Numerator, int Denominator);
